securebanking-openbanking-uk-rs-simulator/src/main/java/com/forgerock/securebanking/openbanking/uk/rs/api/obie/payment/v3_0/domesticpayments/DomesticPaymentsApiController.java

```diff
@@ -0,0 +1,145 @@
+import static com.forgerock.securebanking.openbanking.uk.rs.converter.payment.FRWriteDomesticConsentConverter.toOBDomestic1;
+import static com.forgerock.securebanking.openbanking.uk.rs.converter.payment.FRWriteDomesticConverter.toFRWriteDomestic;
+import static com.forgerock.securebanking.openbanking.uk.rs.persistence.document.payment.FRSubmissionStatus.PENDING;
+
+@Controller("DomesticPaymentsApiV3.0")
+@Slf4j
+public class DomesticPaymentsApiController implements DomesticPaymentsApi {
+
+    private final DomesticPaymentSubmissionRepository domesticPaymentSubmissionRepository;
+    private final IdempotencyValidator idempotencyValidator;
+    private final OBRisk1Validator riskValidator;
+
+    public DomesticPaymentsApiController(DomesticPaymentSubmissionRepository domesticPaymentSubmissionRepository,
+                                         IdempotencyValidator idempotencyValidator,
+                                         OBRisk1Validator riskValidator) {
+        this.domesticPaymentSubmissionRepository = domesticPaymentSubmissionRepository;
+        this.idempotencyValidator = idempotencyValidator;
+        this.riskValidator = riskValidator;
+    }
+
+    @Override
+    public ResponseEntity<OBWriteDomesticResponse1> createDomesticPayments(@Valid OBWriteDomestic1 obWriteDomestic1,
+                                                                           String xFapiFinancialId,
+                                                                           String authorization,
+                                                                           String xIdempotencyKey,
+                                                                           String xJwsSignature,
+                                                                           DateTime xFapiCustomerLastLoggedTime,
+                                                                           String xFapiCustomerIpAddress,
+                                                                           String xFapiInteractionId,
+                                                                           String xCustomerUserAgent,
+                                                                           HttpServletRequest request,
+                                                                           Principal principal
+    ) throws OBErrorResponseException {
+        log.debug("Received payment submission: '{}'", obWriteDomestic1);
+
+        try {
+            // TODO - before we get this far, the IG will need to:
+            //      - verify the consent status
+            //      - verify the payment details match those in the payment consent
+            //      - verify security concerns (e.g. detached JWS, access token, roles, MTLS etc.)
+            idempotencyValidator.verifyIdempotencyKeyLength(xIdempotencyKey);
+            riskValidator.validate(obWriteDomestic1.getRisk());
```

Comment:
This is the only validation the simulator is doing. Note the list in the TODO statement that the IG will need to do.